text/0000-rust-llm.md

```diff
@@ -0,0 +1,78 @@
+# RFC: Add large language models to Rust
+
+This adds a new module to the standard library containing interfaces to large language models.
+
+The author highlights that no large language models were used in the making of this RFC exactly because Rust does not support them today.
+
+## Motivation
+
+Large language models (LLMs for short) have become ubiquitous.
+But using AI language models in Rust has been a lot harder than one would expect.
+The standard library provides neither an interface to nor an implementation of a large language model.
+This will significantly hurt adoption in the future as Rust continues to fall behind other ecosystems, like NodeJS built-in `node:llm` or C23's `llm.h`.
+To ensure continued adoption of Rust in enterprise contexts, this RFC proposes enhancing Rust by adding first-class large language model capabilities to its standard library.
+
+## Guide-level explanation
+
+This RFC contains the following changes
+
+- add `std::llm`, containing several interfaces and potentially an implementation of a large language model
+- add a `cargo llm` subcommand for interfacing with `std::llm` from the command line.
+
+The precise API of the standard library model has not yet been defined and should be evaluated by a relevant committee.
+We propose repurposing the existing WG-llvm for this, as it is lexicographically the closest.
+LLVM backend maintenance will be carried out by a large language model in the meantime.
+WG-llvm is believed to have enough capacity for this task, but this should be confirmed with the working group before accepting this RFC.
+
+## Reference-level explanation
+
+The standard library is enhanced by an additional module, `std::llm`.
+This module is part of the `std` crate and not available in `#![no_std]`, as it requires interfacing with an operating system (the large language model, which is basically an operating system).
+The module contains advanced functionality for interfacing with large language models.
+In the initial implementation, we propose adding support for the following large language models:
+- ChatGPT 3.5
+- ChatGPT 4
+- Google Gemini
+- Clang
+
+The ChatGPT language model will be invoked with the existing `std::request` functionality.
+The Rust Foundation can provide an OpenAI account with credentials injected into the precompiled standard library,
+though the security implications of this should be evaluated with the Rust Foundation security engineer first.
+This enhances user experience because the user does not have to set up an account themselves. The integration with Gemini has lot been decided yet.
```

Comment:
s/lot/not

Reply:
I would have loved to use an LLM, but as mentioned in the opening paragraphs, this was not possible, as Rust does not support them yet.